### PR TITLE
Generate a form for a shipment

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Add function to generate a form for a shipment
- Add unit test and cassette

# Testing

- [x] New unit test and cassette
- [x] All other tests pass as expected 

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Comment by nwithan8** [comment]
Thanks for the second (and third) pair of eyes @Justintime50 @jchen293

File: src/resources/shipment.js [hunk 1]
```diff
@@ -169,6 +169,31 @@ export default (api) =>
       return this.rpc('insure', { amount });
     }
 
+    /**
+     * Generate a form for a shipment.
+     * @param {string} formType
+     * @param formOptions
```
Comment (Justintime50) [inline src/resources/shipment.js:175]: Nit: Should be object type

File: src/resources/shipment.js [hunk 2]
```diff
@@ -169,6 +169,31 @@ export default (api) =>
...
+     * @returns {this}
+     */
+    async generateForm(formType, formOptions) {
+      // eslint-disable-next-line no-param-reassign
+      formOptions = formOptions || {};
```
Comment (Justintime50) [inline src/resources/shipment.js:180]: Issue: Variable reassignment can be dangerous which is why the linter is angry it's happening. Let's not ignore it but follow the linter - it's here to help.

**Review by Justintime50 — CHANGES_REQUESTED** [review]

**Review by jchen293 — COMMENTED** [review]
Do we need to add a typescript definition for the `generateForm` function in this PR?

File: src/resources/shipment.js (hunk 2 above)
Comment (nwithan8) [reply src/resources/shipment.js:180]: Resolved

File: src/resources/shipment.js (hunk 1 above)
Comment (nwithan8) [reply src/resources/shipment.js:175]: Resolved

File: src/resources/shipment.js [hunk 3]
```diff
@@ -169,6 +169,30 @@ export default (api) =>
...
     }
 
+    /**
+     * Generate a form for a shipment.
+     * @param {string} formType
+     * @param {object} formOptions
+     * @returns {this}
+     */
+    async generateForm(formType, formOptions) {
+      const _formOptions = formOptions || {};
```
Comment (jchen293) [inline src/resources/shipment.js:179]:
```suggestion
    async generateForm(formType, formOptions = {}) {
      const _formOptions = formOptions || {};
```

I think you can make `formOptions` as an optional parameter by doing this way

File: types/Shipment/Shipment.d.ts [hunk 4]
```diff
@@ -288,4 +288,12 @@ export declare class Shipment implements IShipment {
...
    */
   public insure(amount: number): Promise<Shipment>;
+
+  /**
+   * Generate a specific type of form for the Shipment, using the provided form options.
+   *
+   * @param formType The type of form to generate.
+   * @param formOptions Additional options to use when generating the form.
+   */
+  public generateForm(formType: string, formOptions: object | null): Promise<Shipment>;
```
Comment (jchen293) [inline types/Shipment/Shipment.d.ts:298]:
```suggestion
  public generateForm(formType: string, formOptions?: object): Promise<Shipment>;
```

Use `?` to indicate the parameter variable is optional.

https://www.educative.io/answers/how-to-use-a-question-mark-in-typescript-variables

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**